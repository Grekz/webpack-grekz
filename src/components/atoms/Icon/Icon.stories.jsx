import React from 'react'

import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'

import Icon from './Icon'

storiesOf('Icon', module).add('empty', () => <Icon alt="test" />)
