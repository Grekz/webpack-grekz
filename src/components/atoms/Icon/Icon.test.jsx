import React from 'react'
import Icon from './Icon'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<Icon /> rendering', () => {
  it('should render Icon', () => {
    const wrapper = shallow(<Icon />)
    expect(wrapper).toHaveLength(1)
  })
  it('should contain 1 svg element', () => {
    const wrapper = shallow(<Icon />)
    expect(wrapper.find('svg')).toHaveLength(1)
  })
  it('mathces the snapshot', () => {
    const tree = shallow(<Icon />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
