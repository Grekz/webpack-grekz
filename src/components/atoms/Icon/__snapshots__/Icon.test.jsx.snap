// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> rendering mathces the snapshot 1`] = `
<svg
  height="22"
  viewBox="0 0 1024 1024"
  width="22"
>
  <path />
</svg>
`;
